File: server/src/Models/DMToolkitModels/Instances/CharacterClassInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class CharacterClassInstance : IDefinitionInstance<CharacterClassDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public int Level { get; set; } = 0;
    public List<int> HpRolls { get; set; } = new();
    public ICollection<FeatInstance> FeatInstances { get; set; } = new List<FeatInstance>();
    public string SubclassInstanceId { get; set; } = string.Empty; 
    public SubclassInstance? SubclassInstance { get; set; } = null;
    public bool SelectedItemSet { get; set; } = false;
    public string DefinitionId { get; set; } = string.Empty;
    public CharacterClassDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/characterClassInstance.ts
import type { CharacterClassDefinition } from "../definitions/characterClassDefinition";
import type { FeatInstance, FeatInstanceDto } from "./featInstance";
import type { SubclassInstance, SubclassInstanceDto } from "./subclassInstance";

export interface CharacterClassInstance {
    id: string;
    level: number;
    hpRolls: number[];
    featInstances: FeatInstance[];
    subclassInstance: SubclassInstance | null;
    selectedItemSet: boolean;
    definition: CharacterClassDefinition;
}

export interface CharacterClassInstanceDto {
    id: string;
    level: number;
    hpRolls: number[];
    featInstances: FeatInstanceDto[];
    subclassInstance: SubclassInstanceDto | null;
    selectedItemSet: boolean;
    definitionId: string;
}